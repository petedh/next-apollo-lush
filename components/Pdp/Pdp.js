import styles from './Pdp.module.css'
import parse from 'html-react-parser'
import FourOhFour from '../FourOhFour/FourOhFour'

const Pdp = (product) => {
  if (!product.product) {
    return <FourOhFour />
  } else {
    const {
      product: {
        description,
        media,
        name,
        pricing: {
          priceRange: {
            start: { gross },
          },
        },
        seoTitle,
        thumbnail: { url },
      },
    } = product
    const descMarkup = JSON.parse(description)
    const heroImg = media.length > 0 ? media[1].url : media[0].url
    return (
      <>
        <div className={styles.pdp}>
          <div className={styles.hero}>
            <img
              alt={seoTitle}
              height={1000}
              src={heroImg}
              title={seoTitle}
              width={1000}
            />
          </div>
          <div className={styles.details}>
            <div className={styles.detailsHead}>
              <img
                alt={seoTitle}
                height={255}
                src={url}
                title={seoTitle}
                width={255}
              />
              <h1>{name}</h1>
              <h2>Price: Â£{gross.amount.toFixed(2)}</h2>
            </div>
            {descMarkup.blocks.map((block, index) => (
              <div key={`desc desc-${index}`}>{parse(block.data.text)}</div>
            ))}
          </div>
        </div>
      </>
    )
  }
}
export default Pdp
