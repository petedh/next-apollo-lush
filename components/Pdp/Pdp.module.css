@value small, med from "../../styles/breakpoints.module.css";

.pdp {
  display: grid;
  grid-template-columns: 1fr;
  gap: 2.5rem;
  grid-auto-flow: row;
}

.hero img {
  width: 100%;
  height: auto;
}

.detailsHead {
  text-align: center;
  margin-bottom: 50px;
}

@media small {
  .pdp {
    grid-template-columns: 3fr 2fr;
  }

  .hero img {
    margin-left: -100px;
  }
}

@media med {
  .detailsHead {
    margin: 50px;
  }
}

.details h1 {
  font-family: handwritten, hn-display, HelveticaNeue, Helvetica, helvetica,
    sans-serif;
  font-size: 3rem;
}

.details b {
  font-family: handwritten, hn-display, HelveticaNeue, Helvetica, helvetica,
    sans-serif;
  font-size: 1.8rem;
  margin-top: 25px;
  display: block;
}

.details s {
  color: rgb(19, 134, 69);
  text-decoration: none;
}
