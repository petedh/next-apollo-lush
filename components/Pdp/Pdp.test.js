import Pdp from './Pdp'
import renderer from 'react-test-renderer'
//import { render } from '@testing-library/react'
import '@testing-library/jest-dom'

describe('PDP', () => {
  it('renders Pdp unchanged', () => {
    const tree = renderer.create(<Pdp product={PdpProps} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders 404 with no product', () => {
    const testRenderer = renderer.create(<Pdp product={null} />)
    const testInstance = testRenderer.root
    expect(testInstance.findByType('h3').toBeTruthy)
  })
})

const PdpProps = {
  __typename: 'Product',
  description:
    '{"blocks": [{"data": {"text": "If you\\u2019re looking for biodegradable face wipes, look no further! The convenience of a face wipe meets the luxury of 9 To 5 cream cleanser, with none of the waste. This reusable, biodegradable cleanser and makeup remover feels gentle, while it works hard to clean, balance and clarify your skin."}, "type": "paragraph"}, {"data": {"text": ""}, "type": "paragraph"}, {"data": {"text": "<b>Benefits:</b>"}, "type": "paragraph"}, {"data": {"text": "- Fresh dove orchid infusion is gently yet deeply cleansing on the skin."}, "type": "paragraph"}, {"data": {"text": "- Carrageenan extract is used to thicken and stabilise the product while providing moisture to your skin."}, "type": "paragraph"}, {"data": {"text": "- Ylang ylang oil adds a floral heady scent and balances skin."}, "type": "paragraph"}, {"data": {"text": "- This biodegradable face wipe combines convenience with a conscience. Packaging free and great for travel, this cleanser helps to save unnecessary waste ending up in landfill (or the oceans!)."}, "type": "paragraph"}, {"data": {"text": ""}, "type": "paragraph"}, {"data": {"text": "<b>How to use: </b>"}, "type": "paragraph"}, {"data": {"text": "Simply splash your face with water, wet one side of the pad, and glide over your face to activate a light cleansing cream that can be gently worked into the skin and rinsed away. Remove light eye makeup by sweeping the wet cleansing wipe over your eyelids and rinsing away to gently rub away mascara. Cleansing complete, you can then finish your routine as usual."}, "type": "paragraph"}, {"data": {"text": "Follow with other soothing products such as Angels on Bare Skin fresh cleanser and Celestial facial moisturiser to complete your soothing and softening skincare routine."}, "type": "paragraph"}, {"data": {"text": ""}, "type": "paragraph"}, {"data": {"text": "<b>How to store: </b>"}, "type": "paragraph"}, {"data": {"text": "Store and transport in a soap dish or Lush tin. Keep in a cool and dry place to use again and again."}, "type": "paragraph"}, {"data": {"text": ""}, "type": "paragraph"}, {"data": {"text": "<b>List of ingredients</b>"}, "type": "paragraph"}, {"data": {"text": ""}, "type": "paragraph"}, {"data": {"text": "<s>Fresh Dove Orchid Infusion (Dendrobium), Talc, Glycerine, </s>Sodium Lauroyl Sarcosinate, Propylene Glycol, <s>Ground Cannellini Beans (Phaseolus vulgaris), Carrageenan Extract (Chondrus crispus), Everlasting Flower Absolute (Helichrysum italicum), Ylang Ylang Oil (Cananga odorata), </s>Alpha-Isomethyl ionone, Amyl Cinnamal, Benzyl Alcohol, <s>*Benzyl Benzoate,</s> Butylphenyl Methylpropional, Cinnamyl Alcohol, Citronellol, Geraniol, Hydroxycitronellal, <s>*Linalool,</s> Perfume"}, "type": "paragraph"}, {"data": {"text": ""}, "type": "paragraph"}, {"data": {"text": "<s> \\u2022  Natural Ingredients</s>"}, "type": "paragraph"}, {"data": {"text": " \\u2022  Safe Synthetics"}, "type": "paragraph"}, {"data": {"text": ""}, "type": "paragraph"}, {"data": {"text": "*occurs naturally in essential oils."}, "type": "paragraph"}]}',
  id: 'UHJvZHVjdDoxNzY=',
  name: '7 to 3',
  pricing: {
    __typename: 'ProductPricingInfo',
    priceRange: {
      __typename: 'TaxedMoneyRange',
      start: {
        __typename: 'TaxedMoney',
        gross: {
          __typename: 'Money',
          amount: 2,
          currency: 'GBP',
        },
      },
      stop: {
        __typename: 'TaxedMoney',
        gross: {
          __typename: 'Money',
          amount: 2,
          currency: 'GBP',
        },
      },
    },
  },
  seoTitle: '7 to 3 Cleansing Wipe',
  slug: '7-to-3',
  thumbnail: {
    __typename: 'Image',
    url: 'https://twstg2.eu.saleor.cloud/media/__sized__/products/7_to_3_cleansing_wipe_2020-thumbnail-255x255.png',
  },
}
