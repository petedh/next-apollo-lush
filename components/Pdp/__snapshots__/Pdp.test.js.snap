// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PDP renders Pdp unchanged 1`] = `
<div
  className="pdp"
>
  <div
    className="hero"
  >
    <img
      alt="7 to 3 Cleansing Wipe"
      height={1000}
      src="https://twstg2.eu.saleor.cloud/media/products/12_days_of_christmas_gift_hero_image_2020_01.jpg"
      title="7 to 3 Cleansing Wipe"
      width={1000}
    />
  </div>
  <div
    className="details"
  >
    <div
      className="detailsHead"
    >
      <img
        alt="7 to 3 Cleansing Wipe"
        height={255}
        src="https://twstg2.eu.saleor.cloud/media/__sized__/products/7_to_3_cleansing_wipe_2020-thumbnail-255x255.png"
        title="7 to 3 Cleansing Wipe"
        width={255}
      />
      <h1>
        7 to 3
      </h1>
      <h2>
        Price: £
        2.00
      </h2>
    </div>
    <div>
      If you’re looking for biodegradable face wipes, look no further! The convenience of a face wipe meets the luxury of 9 To 5 cream cleanser, with none of the waste. This reusable, biodegradable cleanser and makeup remover feels gentle, while it works hard to clean, balance and clarify your skin.
    </div>
    <div />
    <div>
      <b>
        Benefits:
      </b>
    </div>
    <div>
      - Fresh dove orchid infusion is gently yet deeply cleansing on the skin.
    </div>
    <div>
      - Carrageenan extract is used to thicken and stabilise the product while providing moisture to your skin.
    </div>
    <div>
      - Ylang ylang oil adds a floral heady scent and balances skin.
    </div>
    <div>
      - This biodegradable face wipe combines convenience with a conscience. Packaging free and great for travel, this cleanser helps to save unnecessary waste ending up in landfill (or the oceans!).
    </div>
    <div />
    <div>
      <b>
        How to use: 
      </b>
    </div>
    <div>
      Simply splash your face with water, wet one side of the pad, and glide over your face to activate a light cleansing cream that can be gently worked into the skin and rinsed away. Remove light eye makeup by sweeping the wet cleansing wipe over your eyelids and rinsing away to gently rub away mascara. Cleansing complete, you can then finish your routine as usual.
    </div>
    <div>
      Follow with other soothing products such as Angels on Bare Skin fresh cleanser and Celestial facial moisturiser to complete your soothing and softening skincare routine.
    </div>
    <div />
    <div>
      <b>
        How to store: 
      </b>
    </div>
    <div>
      Store and transport in a soap dish or Lush tin. Keep in a cool and dry place to use again and again.
    </div>
    <div />
    <div>
      <b>
        List of ingredients
      </b>
    </div>
    <div />
    <div>
      <s>
        Fresh Dove Orchid Infusion (Dendrobium), Talc, Glycerine, 
      </s>
      Sodium Lauroyl Sarcosinate, Propylene Glycol, 
      <s>
        Ground Cannellini Beans (Phaseolus vulgaris), Carrageenan Extract (Chondrus crispus), Everlasting Flower Absolute (Helichrysum italicum), Ylang Ylang Oil (Cananga odorata), 
      </s>
      Alpha-Isomethyl ionone, Amyl Cinnamal, Benzyl Alcohol, 
      <s>
        *Benzyl Benzoate,
      </s>
       Butylphenyl Methylpropional, Cinnamyl Alcohol, Citronellol, Geraniol, Hydroxycitronellal, 
      <s>
        *Linalool,
      </s>
       Perfume
    </div>
    <div />
    <div>
      <s>
         •  Natural Ingredients
      </s>
    </div>
    <div>
       •  Safe Synthetics
    </div>
    <div />
    <div>
      *occurs naturally in essential oils.
    </div>
  </div>
</div>
`;
